internal/handlers/pg/pgdb/query_iterator.go

```diff
@@ -42,16 +42,12 @@ type queryIterator struct {
 func newIterator(ctx context.Context, rows pgx.Rows) iterator.Interface[uint32, *types.Document] {
 	// queryIterator is defined as pointer to address it in the finalizer.
 	qi := &queryIterator{
-		ctx:      ctx,
-		rows:     rows,
-		closed:   false,
-		mxClosed: new(sync.Mutex),
+		ctx:  ctx,
+		rows: rows,
 	}
 
 	runtime.SetFinalizer(qi, func(qi *queryIterator) {
-		qi.mxClosed.Lock()
-		defer qi.mxClosed.Unlock()
-		if !qi.closed {
+		if !qi.closed.Load() {
```

Comment:
```suggestion
		it.close.Do(func() {
```

No need to use a separate flag. And please call it `it` everywhere for consistency

Reply:
Ah!